src/software/ai/passing/cost_function.cpp

```diff
@@ -11,159 +11,128 @@
     }
 
-    // Figure out what the maximum open angle of the goal could be from the receiver pos.
-    Angle goal_angle = convexAngle(field.enemyGoalpostNeg(), pass.receiverPoint(),
-                                   field.enemyGoalpostPos())
-                           .abs();
-    double net_percent_open = 0;
-    if (goal_angle > Angle::zero())
-    {
-        net_percent_open = open_angle_to_goal.toDegrees() / goal_angle.toDegrees();
-    }
+    const double min_ideal_angle =
+        passing_config.min_ideal_pass_shoot_goal_open_angle_deg();
+    double open_angle_to_goal_score = open_angle_to_goal.toDegrees();
 
-    // Create the shoot score by creating a sigmoid that goes to a large value as
-    // the section of net we're shooting on approaches 100% (ie. completely open)
-    double shot_openness_score = sigmoid(net_percent_open, 0.45, 0.95);
-
-    // Prefer angles where the robot does not have to turn much after receiving the
-    // pass to take the shot (or equivalently the shot deflection angle)
-    //
-    // Receiver robots on the friendly side, almost always, need to rotate a full 180
-    // degrees to shoot on net. So we relax that requirement for both receiver and ball
-    // locations on the friendly side
-    //
-    // TODO (#1987) This creates a very steep slope, find a better way to do this
-    if (pass.receiverPoint().x() < 0 || pass.passerPoint().x() < 0)
-    {
-        ideal_max_rotation_to_shoot_degrees = 180;
-    }
-    Angle rotation_to_shot_target_after_pass = pass.receiverOrientation().minDiff(
-        (shot_target - pass.receiverPoint()).orientation());
-    double required_rotation_for_shot_score =
-        1 - sigmoid(rotation_to_shot_target_after_pass.abs().toDegrees(),
-                    ideal_max_rotation_to_shoot_degrees, 4);
+    // Clamp angle to [0, min_ideal_angle], where all angle >=min_ideal_angle are given
+    // a score of 1.0.
+    open_angle_to_goal_score = std::clamp(open_angle_to_goal_score, 0.0, min_ideal_angle);
+
+    // Linearly scale score to [0.0, 1.0]
+    open_angle_to_goal_score = open_angle_to_goal_score / min_ideal_angle;
+
+    // Linearly scale score to [min_pass_shoot_score, 1.0] to stop this cost function
+    // from returning a very low score, causing the other cost functions to be ignored.
+    return open_angle_to_goal_score;
+}
+
+double ratePassShootScore(const Field& field, const Team& enemy_team, const Pass& pass,
+                          const TbotsProto::PassingConfig& passing_config)
+{
+    double shot_score = rateShot(pass.receiverPoint(), field, enemy_team, passing_config);
 
-    return shot_openness_score * required_rotation_for_shot_score;
+    // Linearly scale score to [min_pass_shoot_score, 1.0] to stop this cost function
+    // from returning a very low score, causing the other cost functions to be ignored.
+    return normalizeValueToRange(shot_score, 0.0, 1.0,
+                                 passing_config.min_pass_shoot_score(), 1.0);
 }
```

Comment:
`ratePassShootScore` historically kills the score of many passes to 0, and stops us from being able to pass to other robots without the sole intention of shooting on net afterwards. So the score is normalized to [0.5, 1.0] to incentivize passes to positions with better shooting ability, but not have it zero out scores.